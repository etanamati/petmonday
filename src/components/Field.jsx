import React from 'react';
import {Form} from 'react-bootstrap';
import PropTypes from 'prop-types';

const Field = ({ id, label, help, ...props }) => {
  return (
    <Form.Group controlId={id} >
      <Form.Label>{label}</Form.Label>
      <Form.Control {...props} />
      {help && <Form.HelpBlock>{help}</Form.HelpBlock>}
    </Form.Group>
  );
}

Field.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string
}

export default Field;