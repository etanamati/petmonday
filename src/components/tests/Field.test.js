import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Field from '../Field';

it('should render app', () => {
  const wrapper = shallow(<Field id='1' label='Label' help='Help'/>);
    
  expect(toJson(wrapper)).toMatchSnapshot();
});