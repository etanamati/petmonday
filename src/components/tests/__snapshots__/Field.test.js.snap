// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render app 1`] = `
<ForwardRef(Bootstrap(FormGroup))
  controlId="1"
>
  <ForwardRef(ContextTransform)>
    Label
  </ForwardRef(ContextTransform)>
  <ForwardRef(ContextTransform) />
  <Component>
    Help
  </Component>
</ForwardRef(Bootstrap(FormGroup))>
`;
